### PR TITLE
Add title

### PR DESCRIPTION
**Pull request by sainathr888**
Add description.

### CODE REVIEW

File: src/main/java/com/lowcd/TCS/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import com.lowcd.TCS.entity.User;
+import com.lowcd.TCS.model.UserBO;
+import com.lowcd.TCS.repository.RoleRepository;
+import com.lowcd.TCS.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@CrossOrigin
+@RequestMapping("api/v1/user")
```
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/controller/UserController.java]: use `/user` here

File: src/main/java/com/lowcd/TCS/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Autowired
+    private UserService userservice;
+    @Autowired
+    private RoleRepository roleRepository;
+
+  @PostMapping(path="/save")
+  public String saveUser(@RequestBody UserBO userBO)
+  {
+
```
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/controller/UserController.java:33]: add logic to check if a user already exists with the given mail, if already exists then send the respective response

Comment (srivatsav06) [reply src/main/java/com/lowcd/TCS/controller/UserController.java:33]: and in user model, add unique propertie to email

File: src/main/java/com/lowcd/TCS/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+public class UserController {
+
+    @Autowired
+    private UserService userservice;
+    @Autowired
+    private RoleRepository roleRepository;
+
+  @PostMapping(path="/save")
+  public String saveUser(@RequestBody UserBO userBO)
```
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/controller/UserController.java]: uses return type of `ResponseEntity<>` . refer @RaHuL-5-5-5  recent commit. same applies for login api, and return appropriate HTTP status.

File: src/main/java/com/lowcd/TCS/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      );
+
+      String id = userservice.addUser(user1);
+      return id;
+  }
+
+    @PostMapping(path="/login")
+    public String loginemployee(@RequestBody LoginBO loginuser)
+    {
+        String st = userservice.loginEmployee(loginuser);
```
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/controller/UserController.java:48]: `loginUser` instead of login`loginEmployee`

Comment (srivatsav06) [reply src/main/java/com/lowcd/TCS/controller/UserController.java:48]: `st` is redudent here can directly return `userservice.loginUser()`

File: src/main/java/com/lowcd/TCS/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    private RoleRepository roleRepository;
+
+  @PostMapping(path="/save")
+  public String saveUser(@RequestBody UserBO userBO)
+  {
+
+
+      Role role = roleRepository.findById(userBO.getRoleId()).get();
+      User user1 = new User(
+              userBO.getUserid(),
```
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/controller/UserController.java]: API call should word without taking the id as input

File: src/main/java/com/lowcd/TCS/model/LoginBO.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class LoginBO {
+
+    private  String email;
+    private String password;
+
```
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/model/LoginBO.java:14]: remove extra spaces

File: src/main/java/com/lowcd/TCS/model/UserBO.java
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/model/UserBO.java:1]: remove extra spaces

File: src/main/java/com/lowcd/TCS/service/UserService.java
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/service/UserService.java:1]: remvoe extra spaces

File: src/main/java/com/lowcd/TCS/service/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+@Service
+public class UserServiceImpl implements UserService {
+
+
+    @Autowired
+    private UserRepository userRepository;
+    @Override
+    public String addUser(User user) {
+
+       User user1=new User(
```
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/service/UserServiceImpl.java]: As i have mentioned above, make the changes to this method

File: src/main/java/com/lowcd/TCS/service/UserServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        userRepository.save(user1);
+       return user1.getName();
+    }
+
+    @Override
+    public String loginEmployee(LoginBO loginuser) {
+
+        User user = userRepository.findByEmail(loginuser.getEmail());
+
+        if(user!=null)
```
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/service/UserServiceImpl.java]: use ternary operator here `(condition) ? value1 : value 2;`

File: src/main/java/com/lowcd/TCS/controller/UserController.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+              userBO.getName(),
+              userBO.getPassword(),
+              userBO.getEmail(),
+              role
+      );
+
+      String id = userservice.addUser(user1);
```
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/controller/UserController.java]: change the logic of `addUser` such that it accepts a `UserBO` object and creates a `User` object in the method and then saves it

**Review by srivatsav06 — CHANGES_REQUESTED** [review]
can u do the suggested changes.

File: src/main/java/com/lowcd/TCS/service/UserServiceImpl.java [hunk 10]
```diff
@@ -1,43 +1,49 @@
...
-               user.getEmail(),
-               user.getRole()
-
-
-       );
+    @Override
+    public String addUser(UserBO userBO) {
+//
+//        User user1=new User(
+//                user.getName(),
```
Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/service/UserServiceImpl.java]: remove comments

Comment (srivatsav06) [inline src/main/java/com/lowcd/TCS/service/UserServiceImpl.java]: remove comments

**Review by srivatsav06 — CHANGES_REQUESTED** [review]
minor chnages required

**Review by RaHuL-5-5-5 — APPROVED** [review]

**Review by srivatsav06 — APPROVED** [review]

**Pull request merged**